src/ppx/xr_component.cpp

```diff
@@ -48,6 +50,44 @@ static XrBool32 XrDebugUtilsMessengerCallback(XrDebugUtilsMessageSeverityFlagsEX
     return XR_FALSE; /* OpenXR spec suggests always returning false. */
 }
 
+template <size_t N>
+void CharArrayStrCpy(char (&dst)[N], std::string_view src)
+{
+    PPX_ASSERT_MSG(src.length() < N, "source string too large");
+    size_t length = std::min<size_t>(N - 1, src.length());
+    memcpy(dst, src.data(), length);
+    dst[length] = '\0';
+}
+
+float Sign(float x)
+{
+    return (x < 0 ? -1 : (x > 0 ? 1 : 0));
+}
+
+XrVector3f ForwardVector(const XrPosef& controller)
+{
```

Comment:
Could you reuse GLM's quaternion operations for this?
It seems less error prone to reuse quat rot instead of rewriting it with the known factors like this?

```cpp
glm::vec3 unit_forward(0, 0, -1);
glm::quat rotation(q.x, q.y, q.z, q.w);
auto result = rotation * unit_forward;
return XrVector3f(result.x, result.y, result.z);
```

Reply:
Done, although using explicit `glm::rotate` instead of multiplication, the glm definition of multiplication is a bit at odd with math